include/cvc5/cvc5_types.h

```diff
@@ -252,6 +252,59 @@ enum ProofComponent
 };
 /** Writes a proof component identifier to a stream. */
 std::ostream& operator<<(std::ostream& out, ProofComponent pc) CVC5_EXPORT;
+
+/**
+ * Find synthesis targets, used as a argument to Solver::findSynth. These
+ * specify various kinds of terms that can be found by this method.
+ */
+enum FindSynthTarget
+{
+  /**
+   * Find the next term in the enumeration of the target grammar.
+   */
+  FIND_SYNTH_TARGET_ENUM,
```

Comment:
nit: I don't like the naming convention even if it's useful for code suggestions and avoiding name clashes. So, I am leaving some suggestions below that are completely fine to ignore:
- Change `enum FindSynthTarget` to `enum class SynthTarget`. `SynthTarget` is easier to pronounce. Adding `class` to avoid potential name clashes for enum values.
- Change `FIND_SYNTH_TARGET_ENUM` to `ENUM`, `FIND_ENUM`, or `SYNTH_ENUM`. I prefer just `ENUM` to minimize the effects of the first suggestion: `SynthTarget::ENUM` is shorter than `FIND_SYNTH_TARGET_ENUM`.

Reply:
Unfortunately we can't have shorter names since FindSynthTarget is not an `class enum`.  We can't make is a `enum class` due to current limitations in how these enums are automatically exported to other APIs.

For reference, see other enums in this file. Eventually we should improve this, but this is out of scope for now.